pkg/manager/manager.go

```diff
@@ -290,104 +299,255 @@ func (m MediaManager) AddMovieToLibrary(ctx context.Context, request AddMovieReq
 		return err
 	}
+
 	log.Debugw("listed indexers", "count", len(indexers))
 	if len(indexers) == 0 {
 		return errors.New("no indexers available")
 	}
-	indexerIds := make([]int32, len(indexers))
-	for i, indexer := range indexers {
-		indexerIds[i] = indexer.ID
+
+	snapshot := newReconcileSnapshot(indexers, dcs)
+
+	wg.Add(1)
+	go m.ReconcileMissingMovies(ctx, &wg, snapshot)
+
+	wg.Add(1)
+	go m.ReconcileUnreleasedMovies(ctx, &wg, snapshot)
+
+	wg.Wait()
+	return nil
+}
+
+func (m MediaManager) ReconcileMissingMovies(ctx context.Context, wg *sync.WaitGroup, snapshot *ReconcileSnapshot) error {
+	defer wg.Done()
+
+	if snapshot == nil {
+		return fmt.Errorf("snapshot is nil")
 	}
 
 	movies, err := m.storage.ListMoviesByState(ctx, storage.MovieStateMissing)
 	if err != nil {
 		return fmt.Errorf("couldn't list movies during reconcile: %w", err)
 	}
 
-	for _, mov := range movies {
-		if mov.MovieMetadataID == nil || mov.QualityProfileID == 0 || mov.Monitored == 0 {
-			continue
-		}
-
-		if mov.MovieFileID != nil {
-			// TODO: this probably needs to check if the existing file meets the quality cutoff
-			continue
-		}
+	log := logger.FromCtx(ctx)
 
-		det, err := m.storage.GetMovieMetadata(ctx, table.MovieMetadata.ID.EQ(sqlite.Int32(*mov.MovieMetadataID)))
+	for _, movie := range movies {
+		err = m.reconcileMissingMovie(ctx, movie, snapshot)
 		if err != nil {
-			log.Debugw("failed to find movie metadata", "meta_id", *mov.MovieMetadataID)
-			continue
+			log.Warn("failed to reconcile movie", zap.Error(err))
 		}
+	}
 
-		profile, err := m.storage.GetQualityProfile(ctx, int64(mov.QualityProfileID))
-		if err != nil {
-			log.Warnw("failed to find movie qualityprofile", "quality_id", mov.QualityProfileID)
-			continue
-		}
+	return nil
+}
 
-		releases, err := m.SearchIndexers(ctx, indexerIds, MOVIE_CATEGORIES, det.Title)
-		if err != nil {
-			log.Debugw("failed to search indexer", "indexers", indexerIds, zap.Error(err))
-			continue
-		}
+func (m MediaManager) reconcileMissingMovie(ctx context.Context, movie *storage.Movie, snapshot *ReconcileSnapshot) error {
+	log := logger.FromCtx(ctx)
+	log = log.With("movie id", movie.ID)
 
-		log.Debugw("releases for consideration", "releases", len(releases))
-		releases = slices.DeleteFunc(releases, rejectReleaseFunc(ctx, det, profile, protocolsAvailable))
-		log.Debugw("releases after rejection", "releases", len(releases))
-		if len(releases) == 0 {
-			// move on the next movie
-			continue
-		}
+	if movie.MovieMetadataID == nil {
+		log.Warn("movie metadata id is nil, skipping reconcile")
+		return nil
+	}
 
-		slices.SortFunc(releases, sortReleaseFunc())
-		chosenRelease := releases[len(releases)-1]
+	if movie.QualityProfileID == 0 {
+		log.Warn("movie quality profile id is nil, skipping reconcile")
+		return nil
+	}
 
-		log.Infow("found release", "title", chosenRelease.Title, "proto", *chosenRelease.Protocol)
+	if movie.Monitored == 0 {
+		log.Warn("movie is not monitored, skipping reconcile")
```

Comment:
This could potentially be a ton of logs depending on how you manage your library

Reply:
moved to debu